dvc/scm/git/__init__.py

```diff
@@ -162,6 +163,12 @@ def _add_entry_to_gitignore(entry, gitignore):
                 prefix = "" if last == "\n" else "\n"
             fobj.write("{}{}\n".format(prefix, entry))
 
+    @staticmethod
+    def _escape(entry):
+        meta_characters = r"[]!*#"
```

Comment:
Is there some specification that you took this from?

You've also mentioned that pathspec might have this, but it didn't?

Reply:
@efiop , yes, `pathspec` doesn't have an `escape` method.
Those characters were taken from reading `man gitignore` - https://git-scm.com/docs/gitignore , will add a note.

Reply:
Reading there I see that  `#` and `!` only should be escaped if they appear at the start.

Reply:
@Suor , that's true; I test it for `foo#bar` and `foo!bar` and they don't need escaping as you point it out.

But it doesn't affect if you escape it:

```bash
git init
touch 'foo#bar'
touch 'foo!bar'
echo 'foo\#bar' > .gitignore
echo 'foo\!bar' > .gitignore
git status
```

Reply:
I'd prefer to go with over-escaping rather than over-complicating the code